tfsdk/serve_resource_attribute_plan_modifiers_test.go

```diff
@@ -18,12 +18,11 @@ func (rt testServeResourceTypeAttributePlanModifiers) GetSchema(_ context.Contex
 			"name": {
 				Required: true,
 				Type:     types.StringType,
-				// For the purposes of testing, these plan modifiers behave
-				// differently for certain values of the attribute.
-				// By default, they do nothing.
 				PlanModifiers: []AttributePlanModifier{
 					testWarningDiagModifier{},
-					testErrorDiagModifier{},
```

Comment:
Is there a reason we're dropping `testErrorDiagModifier{}`? I see stuff getting shuffled around and cleaned up, which is great, but this looks like a behavior change that I'm not understanding the rationale of.

Reply:
The original implementation of this test was dependent on the value being equal to the `TESTDIAG` string to conditionally perform the behavior of returning an error diagnostic or not. When the `testWarningDiagModifer` and `testErrorDiagModifier` were modified to return a warning/error unconditionally to make them easier to use in other tests (such as list block plan modifier testing), it had the side-effect of causing _this_ testing to always fail with returning that error rather than happening conditionally based on other parts of the test setup.

Happy to change this however.